addons/account/static/src/js/account_report_widgets.js

```diff
@@ -1,604 +1,636 @@
-                        .filter([['id', '=', context_id]]).first().then(function (context) {
-                            var managerModel = new openerp.Model('account.report.footnotes.manager');
-                            managerModel.call('add_footnote', [[parseInt(context.footnotes_manager_id[0])], $("#type").val(), $("#target_id").val(), $("#column").val(), footNoteSeqNum, note]);
-                            $('#footnoteModal').find('form')[0].reset();
-                            $('#footnoteModal').modal('hide');
-                            $("div.page").append(openerp.qweb.render("savedFootNote", {num: footNoteSeqNum, note: note}));
-                        });
-                    });
+openerp.account.ReportWidgets = openerp.Widget.extend({
+    events: {
+        'click .fa-pencil-square': 'clickPencil',
+        'click .fa-pencil': 'clickPencil',
+        'click .oe-account-foldable': 'fold',
+        'click .oe-account-unfoldable': 'unfold',
+        'click .saveFootNote': 'saveFootNote',
+        'click .oe-account-to-amls': 'displayMoveLinesByAccount',
+        'click span.user_type': 'displayMoveLinesByType',
+        'click span.partner_id': 'displayFollowup',
+        'click span.aml': 'displayMoveLine',
+        'click .fa-trash-o': 'rmContent',
+        'click .closeSummary': 'rmContent',
+        'click .oe-account-saved-summary > span': 'editSummary',
+        "change *[name='date_filter']": 'onChangeDateFilter',
+        "change *[name='date_filter_cmp']": 'onChangeCmpDateFilter',
+        "change *[name='date_to']": 'onChangeCmpDateFilter',
+        "change *[name='date_from']": 'onChangeCmpDateFilter',
+        "change *[name='comparison']": 'onChangeComparison',
+        "click input[name='summary']": 'onClickSummary',
+        "click button.saveSummary": 'saveSummary',
+        'click button.saveContent': 'saveContent',
+        'click button#saveFootNote': 'saveFootNote',
+        'click .oe-account-add-footnote': 'footnoteFromDropdown',
+        'click .oe-account-to-graph': 'displayMoveLinesByAccountGraph',
+        'click .oe-account-to-net': 'displayNetTaxLines',
+        'click .oe-account-to-tax': 'displayTaxLines',
+        'click .oe-account-to-bank-statement': 'display_bank_statement',
+        'click .move_line_id': 'onClickAML',
+        'click .to_invoice': 'displayInvoice',
+        'click .to_payment': 'displayPayment',
+    },
+    saveFootNote: function(e) {
+        self = this;
+        var report_name = $(e.target).parents('#footnoteModal').siblings("div.page").attr("class").split(/\s+/)[2];
+        var context_id = $(e.target).parents('#footnoteModal').siblings("div.page").attr("class").split(/\s+/)[3];
+        var note = $("#note").val().replace(/\r?\n/g, '<br />').replace(/\s+/g, ' ');
+        var model = new openerp.Model('account.report.context.common');
+        model.call('get_context_name_by_report_name', [report_name]).then(function (result) {
```

Comment:
it is usually better to add a return statement when performing a call at the end of a function since it will return a deferred and others function might need to wait for this particular function to have finished before doing smth else

return model.call(...)